azurerm/internal/services/compute/tests/orchestrated_virtual_machine_scale_set_resource_test.go

```diff
@@ -0,0 +1,624 @@
+  resource_group_name = azurerm_resource_group.test.name
+
+  platform_fault_domain_count = 5
+  single_placement_group      = true
+
+  zones = ["1"]
+
+  tags = {
+    ENV = "Test"
+  }
+}
+
+resource "azurerm_network_interface" "first" {
+  name                = "acctestnic1-%[2]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = azurerm_subnet.test.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_windows_virtual_machine" "first" {
+  name                = "accVM1-%[3]d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  size                = "Standard_F2"
+  admin_username      = "adminuser"
+  admin_password      = "P@ssw0rd1234!"
+  network_interface_ids = [
+    azurerm_network_interface.first.id,
+  ]
+
+  source_image_reference {
+    publisher = "MicrosoftWindowsServer"
+    offer     = "WindowsServer"
+    sku       = "2016-Datacenter"
+    version   = "latest"
+  }
+
+  os_disk {
+    storage_account_type = "Standard_LRS"
+    caching              = "ReadWrite"
+  }
+
+  virtual_machine_scale_set_id = azurerm_orchestrated_virtual_machine_scale_set.test.id
+}
+
+resource "azurerm_network_interface" "second" {
+  name                = "acctestnic2-%[2]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = azurerm_subnet.test.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_windows_virtual_machine" "second" {
+  name                = "accVM2-%[3]d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  size                = "Standard_F2"
+  admin_username      = "adminuser"
+  admin_password      = "P@ssw0rd1234!"
+  network_interface_ids = [
+    azurerm_network_interface.second.id,
+  ]
+
+  source_image_reference {
+    publisher = "MicrosoftWindowsServer"
+    offer     = "WindowsServer"
+    sku       = "2019-Datacenter"
+    version   = "latest"
+  }
+
+  os_disk {
+    storage_account_type = "Standard_LRS"
+    caching              = "ReadWrite"
+  }
+
+  virtual_machine_scale_set_id = azurerm_orchestrated_virtual_machine_scale_set.test.id
+}
+`, template, data.RandomInteger, data.RandomIntOfLength(8))
+}
+
+func testAccAzureRMOrchestratedVirtualMachineScaleSet_template(data acceptance.TestData) string {
+	// in VMSS VMO mode, the `platform_fault_domain_count` has different acceptable values for different locations,
+	// therefore this location is fixed to EastUS2 to make sure the acceptance test has no issues about this value
+	location := "EastUS2"
```

Comment:
almost all of the regions support 2 - so we can use a dynamic location and set this to 2? https://github.com/MicrosoftDocs/azure-docs/blob/master/includes/managed-disks-common-fault-domain-region-list.md

if necessary we can look to add a data source for this information that looks it up in the API - but for now it looks sufficient to hard-code this to `2`?

Reply:
Well, I just tried that eastus2 only accepts `5`.

Reply:
And I got some feedback from the service team, there is a difference when the VMSS is zonal or not. In this case, when zonal, eastus2 only supports `platform_fault_domain_count = 5`.
Different locations have different zone support list as well.

Reply:
I could check with the service team about this....
But I am afraid that the problem remains: different locations support different list of zones, and the list of zones is subscription dependent.
However I am fine with replacing this hard-coded location with `data.Locations.Primary`, what do you think? If you prefer we move this hard-coded location away, I will do it then

Reply:
And you can use this REST API `https://management.azure.com/subscriptions/{subsID}/providers/Microsoft.Compute/?api-version=2019-07-01` to query the zone mappings in your subscription.
My result is:
```
"zoneMappings": [
                {
                    "location": "East US 2",
                    "zones": [
                        "2",
                        "3",
                        "1"
                    ]
                },
                {
                    "location": "Central US",
                    "zones": [
                        "2",
                        "3",
                        "1"
                    ]
                },
                {
                    "location": "West Europe",
                    "zones": [
                        "2",
                        "3",
                        "1"
                    ]
                },
                {
                    "location": "East US 2 EUAP",
                    "zones": [
                        "2",
                        "3",
                        "1"
                    ]
                },
                {
                    "location": "Central US EUAP",
                    "zones": [
                        "1",
                        "2"
                    ]
                },
                {
                    "location": "France Central",
                    "zones": [
                        "2",
                        "3",
                        "1"
                    ]
                },
                {
                    "location": "Southeast Asia",
                    "zones": [
                        "2",
                        "3",
                        "1"
                    ]
                },
                {
                    "location": "West US 2",
                    "zones": [
                        "2",
                        "3",
                        "1"
                    ]
                },
                {
                    "location": "North Europe",
                    "zones": [
                        "2",
                        "3",
                        "1"
                    ]
                },
                {
                    "location": "East US",
                    "zones": [
                        "2",
                        "3",
                        "1"
                    ]
                },
                {
                    "location": "UK South",
                    "zones": [
                        "2",
                        "3",
                        "1"
                    ]
                },
                {
                    "location": "Japan East",
                    "zones": [
                        "2",
                        "3",
                        "1"
                    ]
                },
                {
                    "location": "Australia East",
                    "zones": []
                },
                {
                    "location": "South Africa North",
                    "zones": []
                },
                {
                    "location": "South Central US",
                    "zones": []
                },
                {
                    "location": "Canada Central",
                    "zones": []
                }
            ],
```